### PR TITLE
[SPARK-47504][SQL] Resolve AbstractDataType simpleStrings for StringTypeCollated

### PR DESCRIPTION
**Pull request by mihailomilosevic2001**
### What changes were proposed in this pull request?
Renaming simpleString in StringTypeAnyCollation. This PR should only be merged after https://github.com/apache/spark/pull/45383 is merged.

### Why are the changes needed?
[SPARK-47296](https://github.com/apache/spark/pull/45422) introduced a change to fail all unsupported functions. Because of this change expected inputTypes in ExpectsInputTypes had to be changed. This change introduced a change on user side which will print "STRING_ANY_COLLATION" in places where before we printed "STRING" when an error occurred. Concretely if we get an input of Int where StringTypeAnyCollation was expected, we will throw this faulty message for users.

### Does this PR introduce _any_ user-facing change?
Yes

### How was this patch tested?
Existing tests were changed back to "STRING" notation.

### Was this patch authored or co-authored using generative AI tooling?
No.

### CODE REVIEW

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/CollationTypeConstraints.scala [hunk 1]
```diff
@@ -53,6 +53,7 @@ abstract class StringTypeCollated extends AbstractDataType {
  * Use StringTypeBinary for expressions supporting only binary collation.
  */
 case object StringTypeBinary extends StringTypeCollated {
+  // TODO: When this AbstractDataType is first used, think of a proper simpleString name
```
Comment (HyukjinKwon) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/CollationTypeConstraints.scala:56]: Mind filing a JIRA, and make this like:

```
// TODO(SPARK-XXXXX): When this AbstractDataType is first used, think of a proper simpleString name
```

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/StringTypeCollated.scala [hunk 2]
```diff
@@ -30,6 +30,7 @@ abstract class StringTypeCollated extends AbstractDataType {
  * Use StringTypeBinary for expressions supporting only binary collation.
  */
 case object StringTypeBinary extends StringTypeCollated {
+  // TODO (SPARK-47504): Rename this AbstractDataType's simpleString when it is first used
```
Comment (cloud-fan) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/StringTypeCollated.scala:33]: can we fix it now as we know it should be `"string"`?

**Review by cloud-fan — APPROVED** [review]

**Comment by mihailom-db** [comment]
@cloud-fan Could we merge this in? So some new PR does not conflict with it again.

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Pull request closed without merging**